test/filetype/embeddedpuppet.vader

```diff
@@ -0,0 +1,22 @@
+Execute (Filetype detection on a new empty file):
+  edit foo.epp
+  AssertEqual &filetype, 'embeddedpuppet'
+
+Given (embeddedpuppet file with leading tag):
+  <% if { %>
+  content
+
+# This is super unclean: it creates a file at the top of the git repository. We
+# have to hope that a file with this name doesn't already exist. However, vader
+# doesn't seem to have the concept of a temporary directory for test runtime.
+Execute (create file and clear filetype):
+  write! test_with_leading_tag.epp
+  set ft=txt
```

Comment:
You could avoid writing a file here by using :file to name the buffer and then rerunning filetype detection.
```
  file test_with_leading_tag.epp
  filetype detect
```

Reply:
wow thanks that's way better than what I wrote and it seems to work for our testing purposes :)